File: SQCQPpack/SNSQP/solver/PlotSCCA.m
function PlotSPS(v,nn)
    n = length(v);
    v = v([1:50*nn (n-150*nn+1):n]);
    figure('Renderer', 'painters', 'Position', [800 500/nn 400 200])
    axes('Position', [0.07 0.09 0.89 0.88]);
    
    pt= plot(1:length(v),zeros(length(v),1), '-', 'LineWidth', 1); 
    
    pt(1).Color='#f26419'; hold on; %0e6fbb
    stem(find(v), v(v ~= 0), 'o-',  'filled','Color','#1c8ddb', 'MarkerSize',6, 'LineWidth', 1);
    
    grid on;
    
    ymin = min(v(v ~= 0));
    ymax = max(v(v ~= 0));
    y    = max(abs(ymin),abs(ymax));
    axis([1 length(v) -1.05*y 1.05*y]);
    
    if nn == 1
        xticks([1, 50, 150, 200]);
        xticklabels({'1','50', '450','500'});
    else
        xticks([1, 250, 750, 1000]);
        xticklabels({'1','250', '2250','2500'});
    end
    
end

File: SQCQPpack/SNSQP/solver/PlotSCCA.py
from __future__ import annotations

from typing import Iterable

import numpy as np


def plot_sps(vector: Iterable[float], block_size: int = 1):
    """Python port of ``PlotSCCA.m``."""

    try:
        import matplotlib.pyplot as plt
    except ImportError as exc:  # pragma: no cover
        raise RuntimeError('matplotlib is required for plotting') from exc

    v = np.asarray(list(vector), dtype=float)
    n = v.size
    keep = np.concatenate([np.arange(min(50 * block_size, n)), np.arange(max(n - 150 * block_size, 0), n)])
    v = v[keep]

    fig, ax = plt.subplots(figsize=(4, 2))
    ax.plot(np.arange(v.size), np.zeros_like(v), '-', linewidth=1, color='#f26419')
    ax.stem(np.flatnonzero(v), v[v != 0], linefmt='C0-', markerfmt='C0o', basefmt='k-')
    ax.grid(True)

    if np.any(v != 0):
        y = np.max(np.abs(v[v != 0]))
    else:
        y = 1.0
    ax.set_ylim(-1.05 * y, 1.05 * y)

    if block_size == 1:
        ax.set_xticks([0, min(49, v.size - 1), min(149, v.size - 1), min(199, v.size - 1)])
        ax.set_xticklabels(['1', '50', '450', '500'])
    else:
        ax.set_xticks([0, min(249, v.size - 1), min(749, v.size - 1), min(999, v.size - 1)])
        ax.set_xticklabels(['1', '250', '2250', '2500'])

    try:
        plt.show()
    except Exception:
        pass

    return fig, ax
